### PR TITLE
Expose get_proc_info_in_range API

### PR DESCRIPTION
**Pull request by am11**
* Revive https://github.com/libunwind/libunwind/pull/187.
* Add docs and a basic test.

### CODE REVIEW

**Comment by am11** [comment]
cc @mikem8361, @AaronRobinsonMSFT, @janvorli. PTAL.

This is upstreaming https://github.com/dotnet/runtime/tree/2a38e20/src/native/external/libunwind_extras/oop, adapting to the format from  PR #299.

The .man file is autogenerated based on .tex LaTeX file. The Description section can use more/better text. @mikem8361 if you could provide additional description for this document, that would be great.

File: src/mi/Lget_proc_info_in_range.c [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#define UNW_LOCAL_ONLY
```
Comment (AaronRobinsonMSFT) [inline src/mi/Lget_proc_info_in_range.c:1]: License?

Comment (am11) [reply src/mi/Lget_proc_info_in_range.c:1]: The local (L) source just include the global (G) ones with guard where I added the header. None of the other L files have the license: `cat src/mi/L*`.

**Review by mikem8361 — APPROVED** [review]

**Comment by am11** [comment]
@bregma, PTAL. CI failures are unrelated to the change.

**Comment by bregma** [comment]
Agreed. No new regressions introduced. LGTM.

**Pull request merged**